backend/resolvers/queries/showQuery.js

```diff
@@ -1,15 +1,50 @@
 import Show from '../../models/show'
+import Entry from '../../models/entry';
+import User from '../../models/user'
+import { UserError } from 'graphql-errors'
+import { ADMIN } from '../../constants'
 
-export function show (_, args, req) {
+export function show(_, args, req) {
   return Show.findById(args.id)
 }
 
-export function shows (_, args, req) {
-  if (args.orderBy) {
+export function shows(_, args, req) {
+  // If only order is desired, show all shows ordered
+  if (args.orderBy && !args.studentUsername) {
     return Show.findAll({
       order: [[args.orderBy.sort, args.orderBy.direction]]
     })
   }
 
+  // If username given, show all shows the student has submitted to
+  if (args.studentUsername) {
+    const isRequestingOwnUser = req.auth.username !== undefined &&
+      req.auth.username === args.studentUsername
+    if (req.auth.type !== ADMIN && !isRequestingOwnUser) {
+      throw new UserError('Permission Denied')
+    }
+    // Get all the shows the student has been on 
+    // (including as group creator), through entries
+    return User.findById(args.studentUsername).then((student) => {
+      return student.getOwnAndGroupEntries()
+        // Find all connected shows
+        .then((entries) => {
+          const showIds = entries.map(entry => entry.showId)
+          // If order doesn't matter just return all the shows
+          if (!args.orderBy) {
+            return Show.findAll({ where: { id: showIds } })
```

Comment:
just wondering, does this automatically do a `WHERE thing IN (a, b, c)` query? I've seen sequelize that uses the $in operator

Reply:
[Looks like it should](https://github.com/sequelize/sequelize/blob/091abb48ea1aa459fb6955536ae499d47845e342/lib/model.js#L1492)

Reply:
yup. see here: http://docs.sequelizejs.com/manual/tutorial/models-usage.html#-findall-search-for-multiple-elements-in-the-database